h/accounts/schemas.py

```diff
@@ -46,11 +49,21 @@ def unique_email(node, value):
 
 def unique_username(node, value):
     """Colander validator that ensures the username does not exist."""
+    exc = colander.Invalid(node, _("This username is already taken."))
     request = node.bindings["request"]
     user = models.User.get_by_username(request.db, value, request.default_authority)
     if user:  # pragma: no cover
-        msg = _("This username is already taken.")
-        raise colander.Invalid(node, msg)
+        raise exc
+
+    if request.db.scalars(
```

Comment:
We could move this to the user service and maybe even combine it with the usage of get_by_username in something like "is_username_avaiable".

Reply:
Could to. I'm actually not sure there's much benefit in moving this code to a service though, I think it'll only be needed in this one place in the registration schema.